File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Local/Source/Tva.Phasors/FNet/ConnectionParameters.vb
'*******************************************************************************************************
'  ConnectionParameters.vb - FNet specific connection parameters
'  Copyright © 2007 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  02/26/2007 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

Imports System.ComponentModel
Imports System.Runtime.Serialization
Imports Tva.Phasors.FNet.Common

Namespace FNet

    <Serializable()> _
    Public Class ConnectionParameters

        Inherits ConnectionParametersBase

        Private m_frameRate As Int16
        Private m_nominalFrequency As LineFrequency
        Private m_stationName As String

        Protected Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)

            ' Deserialize connection parameters
            m_frameRate = info.GetInt16("frameRate")
            m_nominalFrequency = info.GetValue("nominalFrequency", GetType(LineFrequency))
            m_stationName = info.GetString("stationName")

        End Sub

        Public Sub New()

            m_frameRate = DefaultFrameRate
            m_nominalFrequency = DefaultNominalFrequency

        End Sub

        <Category("Optional Connection Parameters"), Description("Configured frame rate for FNET device."), DefaultValue(DefaultFrameRate)> _
        Public Property FrameRate() As Int16
            Get
                Return m_frameRate
            End Get
            Set(ByVal value As Int16)
                m_frameRate = value
            End Set
        End Property

        <Category("Optional Connection Parameters"), Description("Configured nominal frequency for FNET device."), DefaultValue(DefaultNominalFrequency)> _
        Public Property NominalFrequency() As LineFrequency
            Get
                Return m_nominalFrequency
            End Get
            Set(ByVal value As LineFrequency)
                m_nominalFrequency = value
            End Set
        End Property

        <Category("Optional Connection Parameters"), Description("Station name to use for FNET device.")> _
        Public Property StationName() As String
            Get
                Return m_stationName
            End Get
            Set(ByVal value As String)
                m_stationName = value
            End Set
        End Property

        Public Overrides Sub GetObjectData(ByVal info As SerializationInfo, ByVal context As StreamingContext)

            ' Serialize connection parameters
            info.AddValue("frameRate", m_frameRate)
            info.AddValue("nominalFrequency", m_nominalFrequency, GetType(LineFrequency))
            info.AddValue("stationName", m_stationName)

        End Sub

    End Class

End Namespace

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/FNet/ConnectionParameters.cs
//*******************************************************************************************************
//  ConnectionParameters.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  02/26/2007 - James R Carroll & Jian (Ryan) Zuo
//       Generated original version of source code.
//
//*******************************************************************************************************

using System;
using System.ComponentModel;
using System.Runtime.Serialization;
using System.Security.Permissions;

namespace PCS.PhasorProtocols.FNet
{
    /// <summary>
    /// Represents the extra connection parameters required for a connection to a F-NET device.
    /// </summary>
    /// <remarks>
    /// This class is designed to be exposed by a "PropertyGrid" so a UI can request protocol specific connection parameters.
    /// As a result the <see cref="CategoryAttribute"/> and <see cref="DescriptionAttribute"/> elements should be defined for
    /// each of the exposed properties.
    /// </remarks>
    [Serializable()]
    public class ConnectionParameters : ConnectionParametersBase
    {        
        #region [ Members ]

        // Fields
        private Ticks m_timeOffset;
        private short m_frameRate;
        private LineFrequency m_nominalFrequency;
        private string m_stationName;

        #endregion

        #region [ Constructors ]

        /// <summary>
        /// Creates a new <see cref="ConnectionParameters"/>.
        /// </summary>
        public ConnectionParameters()
        {
            m_timeOffset = Common.DefaultTimeOffset;
            m_frameRate = Common.DefaultFrameRate;
            m_nominalFrequency = Common.DefaultNominalFrequency;
            m_stationName = Common.DefaultStationName;
        }

        /// <summary>
        /// Creates a new <see cref="ConfigurationCell"/> from serialization parameters.
        /// </summary>
        /// <param name="info">The <see cref="SerializationInfo"/> with populated with data.</param>
        /// <param name="context">The source <see cref="StreamingContext"/> for this deserialization.</param>
        protected ConnectionParameters(SerializationInfo info, StreamingContext context)
        {
            // Deserialize connection parameters
            m_timeOffset = info.GetInt64("timeOffset");
            m_frameRate = info.GetInt16("frameRate");
            m_nominalFrequency = (LineFrequency)info.GetValue("nominalFrequency", typeof(LineFrequency));
            m_stationName = info.GetString("stationName");
        }

        #endregion

        #region [ Properties ]

        /// <summary>
        /// Gets or sets time offset of the F-NET device in <see cref="Ticks"/>.
        /// </summary>
        /// <remarks>
        /// F-NET devices normally report time in 11 seconds past real-time, this property defines the offset for this this artificial delay.
        /// Note that the parameter value is in ticks to allow a very high-resolution offset;  1 second = 10,000,000 ticks.
        /// </remarks>
        [Category("Optional Connection Parameters"),
        Description("F-NET devices normally report time in 11 seconds past real-time, this parameter adjusts for this artificial delay.  Note parameter is in ticks (1 second = 10,000,000 ticks)."),
        DefaultValue(Common.DefaultTimeOffset)]
        public Ticks TimeOffset
        {
            get
            {
                return m_timeOffset;
            }
            set
            {
                m_timeOffset = value;
            }
        }

        /// <summary>
        /// Gets or sets the configured frame rate for the F-NET device.
        /// </summary>
        /// <remarks>
        /// This is typically set to 10 frames per second.
        /// </remarks>
        [Category("Optional Connection Parameters"),
        Description("Configured frame rate for F-NET device."),
        DefaultValue(Common.DefaultFrameRate)]
        public short FrameRate
        {
            get
            {
                return m_frameRate;
            }
            set
            {
                if (value < 1)
                    m_frameRate = Common.DefaultFrameRate;
                else
                    m_frameRate = value;
            }
        }

        /// <summary>
        /// Gets or sets the nominal <see cref="LineFrequency"/> of this F-NET device.
        /// </summary>
        [Category("Optional Connection Parameters"),
        Description("Configured nominal frequency for F-NET device."),
        DefaultValue(typeof(LineFrequency), "Hz60")]
        public LineFrequency NominalFrequency
        {
            get
            {
                return m_nominalFrequency;
            }
            set
            {
                m_nominalFrequency = value;
            }
        }

        /// <summary>
        /// Gets or sets the station name for the F-NET device.
        /// </summary>
        [Category("Optional Connection Parameters"),
        Description("Station name to use for F-NET device."),
        DefaultValue(Common.DefaultStationName)]
        public string StationName
        {
            get
            {
                return m_stationName;
            }
            set
            {
                m_stationName = value;
            }
        }

        #endregion

        #region [ Methods ]

        /// <summary>
        /// Populates a <see cref="SerializationInfo"/> with the data needed to serialize the target object.
        /// </summary>
        /// <param name="info">The <see cref="SerializationInfo"/> to populate with data.</param>
        /// <param name="context">The destination <see cref="StreamingContext"/> for this serialization.</param>
        [SecurityPermission(SecurityAction.LinkDemand, Flags = SecurityPermissionFlag.SerializationFormatter)]
        public override void GetObjectData(SerializationInfo info, StreamingContext context)
        {
            // Serialize connection parameters
            info.AddValue("timeOffset", (long)m_timeOffset);
            info.AddValue("frameRate", m_frameRate);
            info.AddValue("nominalFrequency", m_nominalFrequency, typeof(LineFrequency));
            info.AddValue("stationName", m_stationName);
        }

        #endregion
    }
}